src/interface/types.rs

```diff
@@ -243,14 +191,11 @@ pub struct ClippedDirent {
 
 pub const CLIPPED_DIRENT_SIZE: u32 = size_of::<interface::ClippedDirent>() as u32;
 
-pub fn get_int(union_argument: Arg) -> Result<i32, i32> {
-    let data = unsafe { union_argument.dispatch_int };
-    let mut type_checker = Arg { dispatch_long: 0 };
-    //turn part of the union into 0xffffffff, but, Rust
-    //does not like just using the hex value so we are forced to use
-    //a value of -1
-    type_checker.dispatch_int = -1;
-    if (unsafe { union_argument.dispatch_long } & !unsafe { type_checker.dispatch_long }) == 0 {
+pub fn get_int(argument: u64) -> Result<i32, i32> {
```

Comment:
I think adding a comment before all these functions explaining the general flow of how type conversion works would be good.